packages/flutter/lib/src/material/text_field.dart

```diff
@@ -582,6 +582,7 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
             isFocused: focusNode.hasFocus,
             isEmpty: controller.value.text.isEmpty,
             child: child,
+            hintTextOverflow: (widget.maxLines == null || widget.maxLines <= 1) ? TextOverflow.ellipsis : TextOverflow.clip,
```

Comment:
I don't think we should specify `hintTextOverflow` in this case; clipping overflow text can be difficult for users to interpret.